.github/workflows/ci.yaml

```diff
@@ -0,0 +1,32 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches:
+      - "main"
+  workflow-dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      -
+        name: Build and push
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/bagelcli:latest
```

Comment:
This is cool, but it does remind me that we need to figure out how / when to start releasing (automatically?) very soon now. `latest` isn't a great tag target for deployment, we'll want to pin specific versions. I'll make an issue for us to figure this out and clarify the deployment process in response to a new docker-build / release

Reply:
https://github.com/neurobagel/project/issues/86